File: test-suite-jsonb-java/src/main/java/example/Book.java
package example;

import io.micronaut.serde.annotation.Serdeable;
import jakarta.json.bind.annotation.JsonbCreator;
import jakarta.json.bind.annotation.JsonbProperty;

@Serdeable // <1>
public class Book {
    private final String title;
    @JsonbProperty("qty") // <2>
    private final int quantity;

    @JsonbCreator // <3>
    public Book(String title, int quantity) {
        this.title = title;
        this.quantity = quantity;
    }

    public String getTitle() {
        return title;
    }

    public int getQuantity() {
        return quantity;
    }
}

File: doc-examples/example-jsonb-kotlin/src/main/kotlin/example/Book.kt
package example

import io.micronaut.serde.annotation.Serdeable
import jakarta.json.bind.annotation.JsonbCreator
import jakarta.json.bind.annotation.JsonbProperty

@Serdeable // <1>
class Book @JsonbCreator constructor( // <3>
    val title: String,
    @field:JsonbProperty("qty") val quantity: Int // <2>
)
